public/js/utils/source-map.js

```diff
@@ -7,11 +7,16 @@ const { getSource } = require("../selectors");
 const { isEnabled } = require("../feature");
 const { isJavaScript } = require("./source");
 const assert = require("./assert");
+const { prefs } = require("./prefs");
 
 import type { Location } from "./actions/types";
 
 let sourceMapRequests = new Map();
 
+function shouldSourceMap() {
+  return isEnabled("sourceMaps") && prefs.sourceMapsEnabled;
```

Comment:
> can load in whatever json

that's a good point. 

> avoid using feature flags for things that should just sit on a branch

we've had this conversation a couple of times, so i'm going to move it to an issue for discussion (out of scope here). It doesn't have to be long lived, but it would be nice to have a couple understood rules of thumb.